plugins/modules/postgresql_alter_system.py

```diff
@@ -0,0 +1,876 @@
+- name: PostgreSQL view pg_settings reference
+  description: Complete reference of the pg_settings view documentation.
+  link: https://www.postgresql.org/docs/current/view-pg-settings.html
+- name: PostgreSQL ALTER SYSTEM command reference
+  description: Complete reference of the ALTER SYSTEM command documentation.
+  link: https://www.postgresql.org/docs/current/sql-altersystem.html
+
+author:
+- Andrew Klychkov (@Andersson007)
+
+extends_documentation_fragment:
+- community.postgresql.postgres
+'''
+
+EXAMPLES = r'''
+- name: Set work_mem
+  community.postgresql.postgresql_alter_system:
+    param: work_mem
+    value: 1024
+
+- name: Reset work_mem
+  community.postgresql.postgresql_alter_system:
+    param: work_mem
+    value: _RESET
+
+- name: Set TimeZone parameter (careful, case sensitive)
+  community.postgresql.postgresql_alter_system:
+    param: TimeZone
+    value: 'Europe/Paris'
+'''
+
+RETURN = r'''
+attrs:
+  description:
+  - Parameter attributes from C(pg_settings)
+    that do not change.
+  returned: success
+  type: dict
+  sample: {
+    'unit': 'kB',
+    'context': 'user',
+    'vartype': 'integer',
+    'min_val': 64,
+    'max_val': 2147483647,
+    'boot_val': 4096,
+  }
+diff:
+  description:
+  - A dictionary the C(before) and C(after) keys.
+  - Each key contains a dictionary of key-value pairs
+    representing changeable columns and values for the parameter
+    obtained from the pg_settings relation.
+  returned: success
+  type: dict
+  sample: {
+    'before': {
+        'setting': 4096,
+        'pending_restart': false
+    },
+    'after': {
+        'setting': 8192,
+        'pending_restart': false,
+    }
+  }
+
+executed_queries:
+  description:
+  - List of executed queries except SELECTs.
+  - In the check mode, returns a query that
+    would be supposed to get executed in the real mode.
+  returned: success
+  type: list
+  elements: str
+  sample: ["ALTER SYSTEM SET shared_preload_libraries = ''"]
+
+restart_required:
+  description:
+  - Indicates if restart of PostgreSQL is required or not.
+  - Added here for convenience. Can be also determined from
+    the diff["after"]["pending_restart"] return value.
+  returned: success
+  type: bool
+  sample: true
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible_collections.community.postgresql.plugins.module_utils.database import \
+    check_input
+from ansible_collections.community.postgresql.plugins.module_utils.postgres import (
+    connect_to_db,
+    ensure_required_libs,
+    get_conn_params,
+    get_server_version,
+    pg_cursor_args,
+    postgres_common_argument_spec,
+)
+
+executed_queries = []
+
+PG_SUPPORTED_VER = 140000
+
+# GUC_LIST_QUOTE parameters list for each version where they changed (from PG_REQ_VER).
+# It is a tuple of tuples as we need to iterate it in order.
+# TODO it was copied here from postgresql_set. After merge, it should be
```

Comment:
If `postgresql_set` is sunset, will that not solve the problem?

Reply:
Could you please elaborate a bit more on this?

Reply:
Do you mean why are we pulling stuff from postgresql_set?
It wasn't me who write this and I'm not sure how important these function are. I should probably change this todo to add something like re-evaluate if we need this and remove it if not.
As for now, i think it's more reliable to have these two functions here. Anyway, there are only two auxiliary functions from there.
Do i understand the question correctly?

Reply:
@andreasscherbaum ah, yes, sorry, that was the point, you're right. if `_set` goes anyway (hopefully), there's no point to move it around. I'll update the comments